### PR TITLE
Year in music Final setup

### PR DESCRIPTION
**Pull request by 07jasjeet**
### In this PR :
1) YimEndgameScreen implemented.
2) Heat Map legend added for YimStatisticsScreen.
3) Share Functionality added.
4) Improved navigation.
5) Fixed bugs.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -108,6 +108,9 @@ dependencies {
     implementation "io.coil-kt:coil-svg:2.2.2"
     kapt 'com.github.bumptech.glide:compiler:4.14.2'
 
+    // Image Zooming Library Compose (Need better alternatives)
+    implementation 'com.github.SmartToolFactory:Compose-Zoom:0.5.0'
```
Comment (akshaaatt) [inline app/build.gradle:112]: I would rather want us to keep our own implementation of this feature until google themselves bring out a native implementation of the feature. I only like third-party libraries when they are a standard and a necessity for our use cases, created by us (the devs) or if there is no way other than the library.

For now, please remove this and only add relevant code to the repo.

File: app/src/main/java/org/listenbrainz/android/presentation/features/yim/YimViewModel.kt [hunk 2]
```diff
@@ -44,9 +44,11 @@ class YimViewModel @Inject constructor(private val repository: YimRepository, @A
     
     private fun getData() {
         viewModelScope.launch {
-            val response = async {  username?.let { repository.getYimData(username = it)}!! }
-            when (response.await().status){
-                Resource.Status.SUCCESS -> yimData.value = response.await()
+            val response = username?.let {
+                    repository.getYimData(username = it)
+                }!!
```
Comment (akshaaatt) [inline app/src/main/java/org/listenbrainz/android/presentation/features/yim/YimViewModel.kt:49]: I am not eyeing too much on the force unwraps here. But could these be avoided in general?

File: app/build.gradle (hunk 1 above)
Comment (07jasjeet) [reply app/build.gradle:112]: Don't worry ill definitely remove this once i have my own iteration ready. Consider this jugaad for now.

File: app/src/main/java/org/listenbrainz/android/presentation/features/yim/YimViewModel.kt (hunk 2 above)
Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/presentation/features/yim/YimViewModel.kt:49]: My bad looks like I'm watching to many youtube videos xD.
Ill inline these type of cases.

Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/presentation/features/yim/YimViewModel.kt]: Done

File: app/build.gradle (hunk 1 above)
Comment (07jasjeet) [reply app/build.gradle]: Done

**Review by akshaaatt — APPROVED** [review]
Awesome work! Totally love it 💯 
Kudos to you for completing this!

File: app/src/main/java/org/listenbrainz/android/presentation/features/yim/screens/YimTopAlbumsScreen.kt [hunk 3]
```diff
@@ -130,10 +130,9 @@ fun YimTopAlbumsScreen(
             // Share Button and next Button
             AnimatedVisibility(visible = animateShareButton) {
                 YimNavigationStation(
-                    isRedTheme = false,
+                    typeOfImage = arrayOf("artists"),
```
[the remark this answers is not in the document]
Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/presentation/features/yim/screens/YimTopAlbumsScreen.kt:133]: Was on my list, must've forgot. Will do

[the remark this answers is not in the document]
Comment (akshaaatt) [reply app/src/main/java/org/listenbrainz/android/presentation/features/yim/screens/YimTopAlbumsScreen.kt:133]: Merging for now. You can make a separate PR to fix this.

[the remark this answers is not in the document]
Comment (07jasjeet) [reply app/src/main/java/org/listenbrainz/android/presentation/features/yim/screens/YimTopAlbumsScreen.kt:133]: Should I write tests for this? I've wanted to learn testing for a while now.

[the remark this answers is not in the document]
Comment (akshaaatt) [reply app/src/main/java/org/listenbrainz/android/presentation/features/yim/screens/YimTopAlbumsScreen.kt:133]: Sure thing! Please do so :)

**Pull request merged**